### PR TITLE
skip servicebinding check when autoscaler is not provided as a service

### PR DESCRIPTION
**Pull request by cdlliuy**
#158862007

### CODE REVIEW

File: src/autoscaler/scalingengine/cmd/scalingengine/scalingengine_test.go [hunk 1]
```diff
@@ -158,13 +158,13 @@ var _ = Describe("Main", func() {
 						Eventually(secondRunner.Session.Buffer, 10*time.Second).Should(gbytes.Say("retry-acquiring-lock"))
 						runner.Interrupt()
 						Eventually(runner.Session.Buffer, 5*time.Second).Should(gbytes.Say("scalingengine.received-interrupt-signal"))
-						Eventually(runner.Session.Buffer, 5*time.Second).Should(gbytes.Say("scalingengine.successfully-released-lock"))
+						//Eventually(runner.Session.Buffer, 5*time.Second).Should(gbytes.Say("scalingengine.successfully-released-lock"))
```
Comment (boyang9527) [inline src/autoscaler/scalingengine/cmd/scalingengine/scalingengine_test.go]: remove if it is not useful

File: src/autoscaler/scalingengine/cmd/scalingengine/scalingengine_test.go [hunk 2]
```diff
@@ -158,13 +158,13 @@ var _ = Describe("Main", func() {
...
 						Eventually(secondRunner.Session.Buffer, 10*time.Second).Should(gbytes.Say("scalingengine.successfully-acquired-lock"))
 					} else {
 						Eventually(runner.Session.Buffer, 10*time.Second).Should(gbytes.Say("retry-acquiring-lock"))
 						secondRunner.Interrupt()
 						Eventually(secondRunner.Session.Buffer, 5*time.Second).Should(gbytes.Say("scalingengine.received-interrupt-signal"))
-						Eventually(secondRunner.Session.Buffer, 5*time.Second).Should(gbytes.Say("scalingengine.successfully-released-lock"))
+						//Eventually(secondRunner.Session.Buffer, 5*time.Second).Should(gbytes.Say("scalingengine.successfully-released-lock"))
```
Comment (boyang9527) [inline src/autoscaler/scalingengine/cmd/scalingengine/scalingengine_test.go]: same as above

File: src/integration/components.go [hunk 3]
```diff
@@ -95,11 +99,12 @@ type APIServerConfig struct {
 	InfoFilePath           string                 `json:"infoFilePath"`
 	CFAPI                  string                 `json:"cfApi"`
 	SkipSSLValidation      bool                   `json:"skipSSLValidation"`
+	ServiceMode            bool                   `json:"serviceMode"`
```
Comment (boyang9527) [inline src/integration/components.go]: why we need this?

Comment (boyang9527) [inline src/integration/components.go]: why we need this?

**Comment by cdlliuy** [comment]
@boyang9527  updated and rebased per comments

**Pull request merged**